src/components/WeatherWidget.vue

```diff
@@ -0,0 +1,55 @@
+<template>
+  <div class="weather-widget">
+    <WeatherWidgetPlace
+      :="weatherData.location"
+      :date="selectedDayData.date ?? 'Today'"
+    />
+    <WeatherWidgetCurrentData :="selectedDayData" />
+    <WeatherWidgetForecast
+      :forecastData="weatherData.forecast"
+      :selectedDay="selectedDay"
+      @handleDayChange="handleDayChange"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import WeatherWidgetPlace from "./WeatherWidgetPlace.vue";
+import WeatherWidgetCurrentData from "./WeatherWidgetCurrentData.vue";
+import WeatherWidgetForecast from "./WeatherWidgetForecast.vue";
+import { computed, ref } from "vue";
+import { mockWeatherResponseData } from "../helpers/mockWeatherDataResponse";
+import { mapWeatherData } from "../utils.ts/dataMappers";
+
+const selectedDay = ref<number | null>(null);
+const weatherData = mapWeatherData(mockWeatherResponseData);
+
+const selectedDayData = computed(() => {
+  if (selectedDay.value === null) {
+    return weatherData.current;
+  }
+
+  return Object.values(weatherData.forecast)[selectedDay.value];
```

Comment:
why we decided weatherData.forecast to be an object? after that we're converting it in 2 places into array, also we select day based on indexes, not keys

I think an array would be a better option here, it's really important not to mess data types, it can become unmaintainable in large scale apps

Reply:
Converting to an array is quite a cheap operation. I would prefer to use an object here since indices tend to change and are unreliable in my opinion. If the key concern is transformation to the array in multiple places then we should convert it once and propagate to other components.

Reply:
yes, we can keep it as an object if it better suits the purpose, but what I wanted to point out is that we have an object, but we're using it like it was an array, converting it here and also in WeatherWidgetForecast component

keeping it as an object would make sense if we leveraged object keys in any way, but here we just using indexes

I think we can simplify it and convert into array just once

Reply:
I am now converting this data to an array in the mapper function and then accessing it using `.find` based on the `selectedDate`. Because of that I am sure I selected correct data and I don't need to convert it later in `WeatherWidgetForecase` to an array. I know accessing data with object and key will be more performant but I think it is not a big deal when we have with only 6 items.

Let me know what you think about that.